### PR TITLE
refactoring of the create value-set API

### PR DESCRIPTION
**Pull request by sizatenquizit**
## Description

refactoring of the create value-set API

## Tickets

https://cdc-nbs.atlassian.net/browse/CNFT2-1658

### CODE REVIEW

File: apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/valueset/ValueSetCreator.java [hunk 1]
```diff
@@ -1,82 +1,66 @@
...
-			id.setClassCd(ValueSetConstants.CREATE_CLASS_CD);
-			id.setCodeSetNm(codeSetName);
-			valueSet.setId(id);
-			// save changes
+			codeGrp.setCodeSetDescTxt(request.valueSetDescription());
+			codeGrp.setCodeSetShortDescTxt(codeShortDescTxt);
+			codeGrp.setCodeSetNm(codeSetName);
+			codeGrp.setVadsValueSetCode(codeSetName);
+			codeGrp.setLdfPicklistIndCd('Y');
+			codeGrp.setPhinStdValInd('N');   // will add all this properties inside the constructor
```
Comment (mpeels) [inline apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/valueset/ValueSetCreator.java:53]: Question: Could the creation of this `CodeSetGroupMetadatum` be moved into the constructor of the `Codeset`? That would put all of the logic for creating the `Codeset` in one place.

File: apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/valueset/ValueSetCreator.java [hunk 2]
```diff
@@ -110,64 +94,14 @@ public long getCodeSetGroupID() {
...
-				request.getValueSetNm().toUpperCase(), 
-				request.getLdfPicklistIndCd(), 
-				request.getValueSetCode(),
-				request.getValueSetTypeCd(), 
-				request.getValueSetOid(), 
-				request.getValueSetStatusCd(),
-				request.getValueSetStatusTime(),
-				request.getParentIsCd(), 
-				Instant.now(), 
+				request.valueSetType(),
```
Comment (mpeels) [inline apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/valueset/ValueSetCreator.java:99]: Suggestion: The valid types in Classic are determined by the following:
```java
codeSql = "select code \"key\", code_desc_txt \"value\" FROM " + NEDSSConstants.SYSTEM_REFERENCE_TABLE + "..code_value_general where code_set_nm='NBS_QUESTION_TYPE' and code<>'SYS'";            
```
source: `SRTMapDAOImpl.java #3908`

Perhaps we could verify that the provided `valueSetType` matches a code found in this value set?

Comment (sizatenquizit) [reply apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/valueset/ValueSetCreator.java]: I found a CodeSet enum that we can use.

File: apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/valueset/ValueSetCreator.java (hunk 1 above)
Comment (sizatenquizit) [reply apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/valueset/ValueSetCreator.java]: To create a codeset, a CodeSetGroupMetadatatum must be created first.

**Review by alaapbharadwaj — APPROVED** [review]

**Pull request merged**